bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergenieHandler.java

```diff
@@ -0,0 +1,179 @@
+                break;
+            case "EG_PROTO_V21":
+                statusOn = V21_STATE_ON;
+                statusOff = V21_STATE_OFF;
+                break;
+            case "EG_PROTO_WLAN":
+                statusOn = WLAN_STATE_ON;
+                statusOff = WLAN_STATE_OFF;
+                break;
+            default:
+                statusOn = STATE_ON;
+                statusOff = STATE_OFF;
+                break;
+        }
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        if (command instanceof RefreshType) {
+            refreshCache.getValue();
+        } else if (command instanceof OnOffType) {
+            final byte[] ctrl = { DONT_SWITCH, DONT_SWITCH, DONT_SWITCH, DONT_SWITCH };
+            final Matcher matcher = CHANNEL_SOCKET.matcher(channelUID.getId());
+
+            try {
+                boolean found = false;
+
+                if (matcher.find()) {
+                    final int index = Integer.parseInt(matcher.group(1)) - 1;
+
+                    if (index >= 0 && index < SOCKET_COUNT) {
+                        ctrl[index] = OnOffType.ON.equals(command) ? SWITCH_ON : SWITCH_OFF;
+                        stateUpdate(energenieSocket.sendCommand(ctrl));
+                        found = true;
+                    }
+                }
+                if (!found) {
+                    logger.debug("Invalid channel id {}, should be value between 1 and {}", channelUID, SOCKET_COUNT);
+                }
+            } catch (final IOException e) {
+                updateStatus(ThingStatus.OFFLINE);
+                logger.debug("Couldn't get I/O for the connection to: {}:{}.", host, TCP_PORT, e);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        final EnergenieConfiguration config = getConfigAs(EnergenieConfiguration.class);
+
+        if (!config.host.isEmpty() && !config.password.isEmpty()) {
+            refreshInterval = EnergenieConfiguration.DEFAULT_REFRESH_INTERVAL;
+            host = config.host;
+            logger.debug("Initializing EnergenieHandler for Host '{}'", config.host);
+            energenieSocket = new EnergenieSocket(config.host, config.password);
+
+            updateStatus(ThingStatus.UNKNOWN);
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refreshState, 5, refreshInterval, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access device , IP-Address or password not set");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("EnergenieHandler disposed.");
+        final ScheduledFuture<?> refreshJob = this.refreshJob;
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    private @Nullable Boolean refreshState() {
```

Comment:
This only returns null or Boolean.TRUE. Why not make this a primitive boolean instead?

Reply:
@Hilbrand can you comment on this. I guess this is in conjunction with the ExpiringCache.

Reply:
changed to primmitive boolean